test/e2e/opentelemetry/opentelemetry.test.ts

```diff
@@ -23,7 +23,9 @@ createNextDescribe(
       await check(async () => {
         const spans = await getTraces()
         const rootSpans = spans.filter((span) => !span.parentId)
-        return String(rootSpans.length)
+        return rootSpans.length >= numberOfRootTraces
+          ? String(numberOfRootTraces)
+          : rootSpans.length
```

Comment:
@ijjk Why exactly was this needed?
I think that this should never happen: `rootSpans.length > numberOfRootTraces`.

Reply:
I've just dug in, and the issue is that `parentId` is not present. This completely breaks OTel. Not sure why `toContainEqual` didn't throw even though `parentId` was missing.

This issue is present only in the production build.
Digging in on what happened.

Reply:
It seems that requests are fired multiple times but only in dev mode.
I'll create a new PR that enforces it can be `>` only in dev mode. I will also make this more explicit.

Reply:
Followup pr: https://github.com/vercel/next.js/pull/47822